### DISCUSSION TITLE
Network Quality Warning in Ant Media Server.

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#4923] [FAQs] [answered]
I am using Ant Media Server as a real-time media server. But sometimes I get the warning “Weak Network Connection” in the player.html, also “Your connection isn't fast enough to play this stream!” in the play.html even if I have a high value of download rate. Why does Ant Media Server generate this warning?
And how can I remove this warning message?

**Comment by yashtandon113** [comment, marked as answer]
It's a great question. Let me explain

### Typical Scenario
Some users complain about this problem by saying I'm having 1Gbps internet speed and I should not get this warning. Yes it can have 1Gbps internet speed but it can still have this warning.

Because the important thing is E2E network quality(QoS). I mean 1Gbps internet speed is the one that ISP provides to reach the internet. On the other hand, if the server is another continent or server does not have bandwidth. One can get this warning again. So it's not only related to your internet connection speed.

### Reasons
Here are some reasons that decrease the network QoS so that you can get this warning

1. Bandwidth between Server and Client.
2. High Packet Loss (<%1)
3. High Round Trip Time (<150ms)
4. High Jitter (<30ms)

### Solution
For the 1, both server side and client side has enough E2E bandwidth. You can also get rid of 2, 3 and 4 by making the server and client as close as possible.

The last but not least, if you're in a WiFi/Mobile network and the access point is not close or network quality is bad in your environment. There will be high packet loss and it may be the cause of the problem. I mean you can even take a look at there is no problem in your local network.

### Monitor the Stats
Ant Media Server provides some statistics that you can monitor. When I investigate the issues, I monitor the statistics. I see that one of the stats has a bad value.

Ideally, jitter should be below 30ms. Packet loss should be no more than 1%, and network latency shouldn’t exceed 150 ms one-way (300 ms return).

Stats can be monitored under the sample player or through JS SDK.
**https://Ant-media-server-domain:5443/WebRTCAppEE/player.html**

![image](https://user-images.githubusercontent.com/82374739/226699010-e1fb5eef-8d55-4f87-8462-501df4b20fda.png)

**How to remove this warning message in play.html?**

You can disable that warning from play.html file.

Open /usr/local/antmedia/webapps/{your APP name}/play.html

Find and comment out below part as below:

`
`
